Add description for nxp,mcux-edma 'dma-requests' and 'dma-channels'

ZhaoxiangJin opened the pull request:
1. Added description for dma-requests and dma-channels in dts/bindings/dma/nxp,mcux-edma.yaml.

2. Fixed dma-requests value in dts/arm/nxp/nxp_mcxn23x_common.dtsi and dts/arm/nxp/nxp_mcxnx4x_common.dtsi. The 'dma-requests' indicates the maximum value of the DMA request sources (slots) index supported by DMAMUX rather than the number of request sources.

iuliana-prodan commented on dts/arm/nxp/nxp_mcxn23x_common.dtsi:
```diff
@@ -498,7 +498,7 @@
 		compatible = "nxp,mcux-edma";
```
This PR LGTM, but it should be split in 2 commits:
commit 1 - adds description for dma-requests and dma-channels;
commit 2 - fixes dma-req values, which should also have a `Fixes` tag since those values were incorrect.

ZhaoxiangJin committed:
dts: nxp,mcux-edma: Added description for properties

Added description for 'dma-requests' and 'dma-channels' in
dts/bindings/dma/nxp,mcux-edma.yaml.

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>

ZhaoxiangJin committed:
dts: nxp_mcxnxxx: Fixes 'dma-requests' incorrect value

Fixes incorrect 'dma-requests' value in
dts/arm/nxp/nxp_mcxn23x_common.dtsi and
dts/arm/nxp/nxp_mcxnx4x_common.dtsi. The
'dma-requests' indicates the maximum value of
the DMA request sources (slots) index supported
by DMAMUX rather than the number of request sources.

Fixes: #97389 (which introduced the incorrect values)

Signed-off-by: Zhaoxiang Jin <Zhaoxiang.Jin_1@nxp.com>
